src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java

```diff
@@ -193,35 +214,39 @@ private static void logMsg(String uniqname, String originalUrl, HttpServletReque
 		M_log.info(String.format(baseString, user, originalUrl));
 	}
 
-	public static String makeApiCall(String... urlChunks){
-		String url = canvasURL + CANVAS_API_VERSION;
-		for (String chunk : urlChunks) {
-			url += chunk;
-		}
-		M_log.info("API call: " + url);
-		HttpResponse response = Utils.executeApiCall(new HttpGet(url));
+	public static ApiResultWrapper makeApiCall(HttpUriRequest clientRequest){
+		HttpResponse response;
+		response = Utils.executeApiCall(clientRequest);
+
 		if (response == null) {
-			M_log.error(String.format("The Api call %s failed with errors", url));
-			return null;
+			String errMsg = String.format("{\"errorMsg\":\"The request %s failed with errors\"}",
+					clientRequest.getURI().toString());
+			return new ApiResultWrapper(Utils.API_UNKNOWN_ERROR, errMsg, "");
 		}
+
 		int statusCode = response.getStatusLine().getStatusCode();
-		if (statusCode != HttpStatus.SC_OK) {
-			M_log.error(String.format("The Api call %s has failed with status code %s", url, statusCode));
-			return null;
-		}
-		String apiResponse;
-		String errMsg = "The API call %s has failed to extract the response due to %s";
+
+		String apiResponse = "";
+		String errMsg = "{\"errorMsg\":\"The request %s has failed to extract the response due to %s\"}";
 		try {
 			apiResponse = EntityUtils.toString(response.getEntity(), "UTF-8");
 		} catch (IOException e) {
-			M_log.error(String.format(errMsg, url, e.getMessage()));
-			return null;
+			return new ApiResultWrapper(Utils.API_EXCEPTION_ERROR,
+					String.format(errMsg, clientRequest.getURI().toString(), e.getMessage()), "");
 		} catch (Exception e) {
-			M_log.error(String.format(errMsg, url, e.getMessage()));
-			return null;
+			return new ApiResultWrapper(Utils.API_EXCEPTION_ERROR,
+					String.format(errMsg, clientRequest.getURI().toString(), e.getMessage()), "");
 		}
 
-		return apiResponse;
+		return new ApiResultWrapper(statusCode, "", apiResponse);
+	}
+
+	public static String urlConstructor(String... urlChunks) {
+		String url = canvasURL + CANVAS_API_VERSION;
```

Comment:
Stringbuffer would be more memory efficient.  Since there is lots of URL construction it may be worth being efficient to avoid later garbage collections.